streams/test-utils/src/main/java/org/apache/kafka/streams/TestOutputTopic.java

```diff
@@ -0,0 +1,200 @@
+        return record.value();
+    }
+
+    /**
+     * Read one Record KeyValue from output topic.
+     *
+     * @return Next output as KeyValue
+     */
+    public KeyValue<K, V> readKeyValue() {
+        final TestRecord<K, V> record = readRecord();
+        return new KeyValue<>(record.key(), record.value());
+    }
+
+    /**
+     * Read one Record from output topic.
+     *
+     * @return Next output as TestRecord
+     */
+    public TestRecord<K, V> readRecord() {
+        return driver.readRecord(topic, keyDeserializer, valueDeserializer);
+    }
+
+    /**
+     * Read output to List.
+     * If the existing key is modified, it can appear twice in output, but replaced in map
+     *
+     * @return Map of output by key
+     */
+    public List<TestRecord<K, V>> readRecordsToList() {
+        final List<TestRecord<K, V>> output = new LinkedList<>();
+        while (!isEmpty()) {
+            output.add(readRecord());
+        }
+        return output;
+    }
+
+
+    /**
+     * Read output to map.
+     * If the existing key is modified, it can appear twice in output, but replaced in map
+     *
+     * @return Map of output by key
+     */
+    public Map<K, V> readKeyValuesToMap() {
+        final Map<K, V> output = new HashMap<>();
+        TestRecord<K, V> outputRow;
+        while (!isEmpty()) {
+            outputRow = readRecord();
+            output.put(outputRow.key(), outputRow.value());
+        }
+        return output;
+    }
+
+    /**
+     * Read all KeyValues from topic to List.
+     *
+     * @return List of output KeyValues
+     */
+    public List<KeyValue<K, V>> readKeyValuesToList() {
+        final List<KeyValue<K, V>> output = new LinkedList<>();
+        KeyValue<K, V> outputRow;
+        while (!isEmpty()) {
+            outputRow = readKeyValue();
+            output.add(outputRow);
+        }
+        return output;
+    }
+
+    /**
+     * Read all values from topic to List.
+     *
+     * @return List of output values
+     */
+    public List<V> readValuesToList() {
+        final List<V> output = new LinkedList<>();
+        V outputValue;
+        while (!isEmpty()) {
+            outputValue = readValue();
+            output.add(outputValue);
+        }
+        return output;
+    }
+
+    /**
+     * Get size of unread record in the topic queue.
+     *
+     * @return size of topic queue
+     */
+    public final long getQueueSize() {
+        return driver.getQueueSize(topic);
+    }
+
+    /**
+     * Verify if the topic queue is empty.
+     *
+     * @return true if no more record in the topic queue
+     */
+    public final boolean isEmpty() {
+        return driver.isEmpty(topic);
+    }
+
+    @Override
+    public String toString() {
+        return new StringJoiner(", ", TestOutputTopic.class.getSimpleName() + "[", "]")
+                .add("topic='" + topic + "'")
+                .add("keyDeserializer=" + keyDeserializer)
```

Comment:
`keyDeserializer.getClass().getSimpleName()` (same for value)

Reply:
changed